includes/Core/Modules/Module.php

```diff
@@ -345,6 +353,9 @@ final protected function execute_data_request( Data_Request $data ) {
 			$restore_defers[] = $this->get_client()->withDefer( true );
 			if ( $this->authentication->get_oauth_client() !== $oauth_client ) {
 				$restore_defers[] = $oauth_client->get_client()->withDefer( true );
+
+				// Set request as using shared credentials if oAuth clients do not match.
+				$this->is_using_shared_credentials = true;
```

Comment:
Since this is only set at this point, it could potentially be left `true` if there was a nested request. We should set it to `false` at the beginning unconditionally to ensure it is only ever set to `true` for the current request. In this case, it might be redundant to also have it in the `finally` block at the end but it shouldn't hurt.

Reply:
Addressed this in #5727, thank you!